### PR TITLE
Mark aggregated condition ready only when No error happened

### PR DESCRIPTION
**Pull request by nan-yu**
Fixes https://github.com/kubernetes-sigs/application/issues/183

### CODE REVIEW

File: controllers/application_controller.go [hunk 1]
```diff
@@ -96,7 +96,7 @@ func (r *ApplicationReconciler) getNewApplicationStatus(ctx context.Context, app
 		Objects: objectStatuses,
 	}
 	newApplicationStatus.ComponentsReady = fmt.Sprintf("%d/%d", countReady, len(objectStatuses))
-	if aggReady && errs != nil {
+	if aggReady && errs == nil {
 		setReadyCondition(newApplicationStatus, "ComponentsReady", "all components ready")
 	} else {
 		setNotReadyCondition(newApplicationStatus, "ComponentsNotReady", "some components not ready")
```
Comment (ant31) [inline controllers/application_controller.go]: is it possible that all components are ready but there's an error ?
I assume yes, the message / status should reflect that or it's still confusing. 

The status would be in error with "some components not ready", as a user I would check which components isn't ready but won't find any.... 
Could we add a separate case:
something like :
```
if errs != nil { 
  "there's an error"
} else if aggReady {
 "all components ready"
} else 
{
 "some components not ready" // 'some' could be replaced with (len(objectStatuses)-countReady)
}
```

Comment (nan-yu) [reply controllers/application_controller.go]: There are two types of conditions: `Ready` and `Error` and we need to set both.
I think we can leverage the `Unknown` status for the `Ready` condition:
```
if errs != nil {
   //set ready condition to unknown
} else if aggReady {
  "all components ready"
} else {
    "x components not ready"
}

if errs != nil {
   setErrorCondition
} else {
   clearErrorCondition
}
```

**Comment by nan-yu** [comment]
Example when errors happened in aggregating statuses:
```
"status": {
    "observedGeneration": 1,
    "conditions": [
      {
        "type": "Ready",
        "status": "Unknown",
        "reason": "ComponentsReadyUnknown",
        "message": "failed to aggregate all components' statuses, check the Error condition for details",
        "lastUpdateTime": "2020-03-06T20:20:29Z",
        "lastTransitionTime": "2020-03-06T20:20:29Z"
      },
      {
        "type": "Error",
        "status": "True",
        "reason": "ErrorSeen",
        "message": "[fetchComponentListResources error, setOwnerRefForResources error, service status error, statefulset status error]",
        "lastUpdateTime": "2020-03-06T20:20:29Z",
        "lastTransitionTime": "2020-03-06T20:20:29Z"
      }
    ],
    "components": [
      {
        "link": "/api/v1/namespaces/default/services/wordpress-mysql-hsvc",
        "name": "wordpress-mysql-hsvc",
        "kind": "Service",
        "status": "Unknown"
      },
      {
        "link": "/api/v1/namespaces/default/services/wordpress-webserver-hsvc",
        "name": "wordpress-webserver-hsvc",
        "kind": "Service",
        "status": "Unknown"
      },
      {
        "link": "/api/v1/namespaces/default/services/wordpress-webserver-svc",
        "name": "wordpress-webserver-svc",
        "kind": "Service",
        "status": "Unknown"
      },
      {
        "link": "/apis/apps/v1/namespaces/default/statefulsets/wordpress-mysql",
        "name": "wordpress-mysql",
        "kind": "StatefulSet",
        "group": "apps",
        "status": "Unknown"
      },
      {
        "link": "/apis/apps/v1/namespaces/default/statefulsets/wordpress-webserver",
        "name": "wordpress-webserver",
        "kind": "StatefulSet",
        "group": "apps",
        "status": "Unknown"
      }
    ],
    "componentsReady": "0/5"
  }
```

**Comment by ant31** [comment]
thank you,
/lgtm
/approve

**Pull request merged**